spec/ParseObject.spec.js

```diff
@@ -2055,4 +2055,33 @@ describe('Parse.Object testing', () => {
     const object = new Parse.Object('CloudCodeIsNew');
     await object.save();
   });
+
+  fit('should can custom objectId', async () => {
+    await reconfigureServer({
+      allowCustomObjectId: true,
+    });
+
+    Parse.allowCustomObjectId = true;
+
+    const body = {
+      name: 'SaveAudio',
+      data: '{"objectId":"66541dbdffead844f1339e3c","createdAt":"2024-05-27T05:44:29.504Z","updatedAt":"2024-05-27T05:44:29.504Z"}',
+    };
+
+    const name = body.name;
+    const data = JSON.parse(body.data);
+
+    const testObject = new Parse.Object(name);
+    testObject.id = data.objectId;
+
+    const entries = Object.entries(data);
```

Comment:
Also here, the loop is unnecessary, and you don't need to set and field of the object at all. This is about setting a custom ID. So all this extra code should be removed.